crates/ruff_linter/src/rules/flake8_bugbear/rules/loop_iterator_mutated.rs

```diff
@@ -0,0 +1,185 @@
+use ruff_diagnostics::Violation;
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{
+    visitor::{self, Visitor},
+    Expr, ExprAttribute, ExprCall, ExprContext, ExprName, ExprSubscript, Stmt, StmtDelete, StmtFor,
+};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::Diagnostic;
+
+static MUTATING_FUNCTIONS: &'static [&'static str] = &[
+    "append", "sort", "reverse", "remove", "clear", "extend", "insert", "pop", "popitem",
+];
+
+/// ## What it does
+/// Checks for mutation of the iterator of a loop in the loop's body
+///
+/// ## Why is this bad?
+/// Changing the structure that is being iterated over will usually lead to
+/// unintended behavior as not all elements will be addressed.
+///
+/// ## Example
+/// ```python
+/// some_list = [1,2,3]
+/// for i in some_list:
+///   some_list.remove(i) # this will lead to not all elements being printed
+///   print(i)
+/// ```
+///
+///
+/// ## References
+/// - [Python documentation: Mutable Sequence Types](https://docs.python.org/3/library/stdtypes.html#typesseq-mutable)
+#[violation]
+pub struct LoopIteratorMutated;
+
+impl Violation for LoopIteratorMutated {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("editing a loop's mutable iterable often leads to unexpected results/bugs")
+    }
+}
+
+fn _to_name_str(node: &Expr) -> String {
```

Comment:
You may want `collect_call_path` for this, we have an abstraction for this exact thing, but it stores structured components (like a vector of the dot-separated pieces) rather than a `String`.

Reply:
Could it be that the function was removed or renamed? I can't find it anymore.